Running functional_tests.sh fails with -k after first run

## Description of the problem, including code/CLI snippet

```bash
# this will pass
./tools/functional_tests.sh -k
# this will fail
./tools/functional_tests.sh -k
```

## Expected Behavior

Both should pass

## Actual Behavior

```
==================================================================================== FAILURES ====================================================================================
_______________________________________________________________________ test_create_issue_note[inprocess] ________________________________________________________________________

gitlab_cli = <function gitlab_cli.<locals>._gitlab_cli at 0x7f99f225c3a0>, issue = <ProjectIssue iid:2>

    def test_create_issue_note(gitlab_cli, issue):
        body = "body"
    
        cmd = [
            "project-issue-note",
            "create",
            "--project-id",
            issue.project_id,
            "--issue-iid",
            issue.id,
            "--body",
            body,
        ]
        ret = gitlab_cli(cmd)
    
>       assert ret.success
E       assert False
E        +  where False = <pytest_console_scripts.RunResult object at 0x7f99f297bf10>.success

tools/functional/test_cli_v4.py:186: AssertionError
------------------------------------------------------------------------------ Captured stdout call ------------------------------------------------------------------------------
# Running console script: gitlab --config-file /tmp/python-gitlab.cfg project-issue-note create --project-id 41 --issue-iid 27 --body body
# Script return code: 1
# Script stdout:

# Script stderr:
Impossible to create object (404: 404 Not found)
```

## Specifications

  - python-gitlab version: `master`
  - API version you are using (v3/v4): `v4`
  - Gitlab server version (or gitlab.com): latest from dockerhub (i.e. whatever functional_tests.sh pulls down.

--- a/tools/functional/test_cli_v4.py
+++ b/tools/functional/test_cli_v4.py
@@ -177,7 +177,7 @@ def test_create_issue_note(gitlab_cli, issue):
         "--project-id",
         issue.project_id,
         "--issue-iid",
-        issue.id,
+        issue.iid,
         "--body",
         body,
     ]

